tests/ops/op_math/test_controlled_op.py

```diff
@@ -649,14 +650,16 @@ def test_do_queue_false(self):
 class TestMatrix:
     """Tests of Controlled.matrix and Controlled.sparse_matrix"""
 
-    def test_raises_error_if_batching(self):
-        """Test a MatrixUndefinedError is raised if the base is batched."""
+    def test_correct_matrix_dimenions_with_batching(self):
+        """Test batching returns a matrix of the correct dimensions"""
         x = np.array([1.0, 2.0, 3.0])
         base = qml.RX(x, 0)
         op = Controlled(base, 1)
 
-        with pytest.raises(qml.operation.MatrixUndefinedError):
-            op.matrix()
+        matrix = op.matrix()
+
+        assert matrix.shape == (3, 4, 4)
+        assert matrix.shape[-2:] == (4, 4)
```

Comment:
Why adding this duplicated test?

Reply:
Hahaha... I have no idea why I did that. Will remove.